Use default editor if not set by environment variable

dmarsic opened the pull request:
Hi @WolfgangSteiner,

I was playing with Issues recently as it fits nicely with my side projects.

When I tried `issues edit <hash>` I'd receive an error because my `EDITOR` envvar was not set. The error is a permissions issue, because then the command is just `system(' /tmp/<tempfile')`. The proposed change is:

When editing issues, try editors in this order:
* $EDITOR environment variable
* git var GIT_EDITOR
* default value in the program, set to 'editor'

Some extra whitespace was automatically removed by my editor.

Can you let me know if you're still actively supporting this project and what is the licence?

Thanks!

WolfgangSteiner commented:
Hi Dom, 

your pull request caught me off guard, because I was not anticipating that anyone besides me would actually use this tool!
Thanks you so much for your interest. I would be glad to merge your pull request. There seems to be one line of code missing in your pull request. Please have a look and update the request at your convenience!

Regards,
Wolfgang

dmarsic commented:
Thanks Wolfgang, you're right. I didn't notice I dropped the line but was missing that functionality and I had already added it in my own follow-up branch :)

I'm not overly familiar with how GitHub merges multiple commits but I imagine you might be able to squash them (if you don't want just fast forward).

I like this code and appreciate the chance to contribute. If I have any future additions I'll let you know.

dmarsic commented:
Hi Wolfgang,

I wanted to check if you still actively maintain the project? I have this PR not merged and some smaller changes (replace `trollop` dependency with `optimist`, some formatting that works better for me) so I wanted to see if you continue to accept PRs.

I'd also like to ask about the license so that I can consider diverging if you choose not to invest more effort into it.

Thanks!
Dom
